tests/phpunit/AbstractSnowflakeTest.php

```diff
@@ -0,0 +1,182 @@
+
+    protected string $dataDir = __DIR__ . '/data';
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        if (!defined('APP_NAME')) {
+            define('APP_NAME', 'ex-db-snowflake');
+        }
+
+        $config = $this->getConfig();
+
+        $config['parameters']['db']['password'] = $config['parameters']['db']['#password'];
+        unset($config['parameters']['db']['#password']);
+        $this->connection = new Connection($config['parameters']['db']);
+
+        $this->connection->query(
+            sprintf('USE SCHEMA %s', QueryBuilder::quoteIdentifier($config['parameters']['db']['schema']))
+        );
+        $this->connection->query(
+            'alter session set client_timestamp_type_mapping=\'timestamp_ntz\''
+        );
+
+        $this->setupTables();
+
+        $fileSystem = new Filesystem();
+        $fileSystem->remove($this->dataDir . '/out');
+        $fileSystem->remove($this->dataDir . '/runAction/config.json');
+        $fileSystem->remove($this->dataDir . '/runAction/out');
+        $fileSystem->remove($this->dataDir . '/connectionAction/config.json');
+        $fileSystem->remove($this->dataDir . '/connectionAction/out');
+        $fileSystem->remove($this->dataDir . '/getTablesAction/config.json');
+        $fileSystem->remove($this->dataDir . '/getTablesAction/out');
+    }
+
+    public function tearDown(): void
+    {
+        $this->connection->disconnect();
+    }
+
+    public function getConfig(string $driver = 'snowflake'): array
+    {
+        $config = parent::getConfig($driver);
+
+        $config['parameters']['db']['schema'] = $this->getEnv($driver, 'DB_SCHEMA');
+        $config['parameters']['db']['warehouse'] = $this->getEnv($driver, 'DB_WAREHOUSE');
+
+        $config['parameters']['extractor_class'] = 'Snowflake';
+        $config['parameters']['tables'][2]['table']['schema'] = $this->getEnv($driver, 'DB_SCHEMA');
+
+        return $config;
+    }
+
+    public function getConfigRow(string $driver = 'snowflake'): array
+    {
+        $config = parent::getConfigRow($driver);
+
+        $config['parameters']['db']['schema'] = $this->getEnv($driver, 'DB_SCHEMA');
+        $config['parameters']['db']['warehouse'] = $this->getEnv($driver, 'DB_WAREHOUSE');
+
+        $config['parameters']['extractor_class'] = 'Snowflake';
+        $config['parameters']['table']['schema'] = $this->getEnv($driver, 'DB_SCHEMA');
+
+        return $config;
+    }
+
+    public function createApplication(array $config, array $state = []): SnowflakeApplication
+    {
+        $logger = new Logger();
+        return new SnowflakeApplication($config, $logger, $state, $this->dataDir);
+    }
+
+    private function setupTables(): void
+    {
+        $this->removeAllTables();
+        $this->createTypesTable();
+        $this->generateTypesRows();
+        $this->createSalesTable();
+        $this->generateSalesRows();
+        $this->createEscapingTable();
+        $this->generateEscapingRows();
```

Comment:
přepsal jsem to tak aby nebylo potřeba SAPI, které nepodporuje různý balíčky který tu byly potřeba